File: RoboLabWP/RoboLabWP/Model/Game/MZCell.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using RoboLabWP.Model.Basic_Types;
//
//  MZCell.cs
//  Maze
//
//  Класс, объект которого хранит данные о клетке лабиринта. Не изменяется после построения лабиринта.

namespace RoboLabWP.Model.Game
{
    public class MZCell
    {
        public bool isEmpty;
        MZByte _walls;

        public MZCell()
        {
            _walls = new MZByte(0);
        }

        public void setWallsValue(MZByte walls)
        {
            _walls = walls;
        }

        public bool hasWallAtDirection(MZDirection direction)
        {
            int result =  (int)(_walls.value()) & (1 << (int) direction);
            return (result != 0 ? true : false);
        }
    }
}

File: RoboLabWP/RoboLabWPComp/Model/Game/MZCell.cpp
#include "pch.h"
#include "MZCell.h"


MZCell::MZCell(void)
{
	_walls = 0;
}

MZCell::MZCell(MZByte walls)
{
	_walls = walls;
}

MZCell::~MZCell(void)
{
}

void MZCell::setWallsValue(MZByte walls)
{
	_walls = walls;
}

bool MZCell::hasWallAtDirection(MZDirection direction)
{
	return _walls & (1 << direction);
}

MZByte MZCell::wallsValue(void)
{
	return _walls;
}
